orchestra/model/component.py

```diff
@@ -46,11 +55,82 @@ def branch(self):
         branch, commit = self.clone.branch()
         return branch
 
+    @property
+    def recursive_hash(self):
+        if self._recursive_hash is None:
+            raise Exception("Accessed recursive_hash before calling resolve_dependencies")
+        return self._recursive_hash
+
+    def resolve_dependencies(self, configuration):
+        if self._recursive_hash is not None:
+            raise Exception("Called resolve_dependencies twice")
+
+        for build in self.builds.values():
+            build.resolve_dependencies(configuration)
+
+        self._recursive_hash = self._compute_recursive_hash()
+
+    def serialize(self):
+        serialized_component = {
+            "license": self.license,
+            "binary_archives": self.binary_archives,
+            "build_from_source": self.build_from_source,
+            "skip_post_install": self.skip_post_install,
+            "add_to_path": self.add_to_path,
+            "repository": self.repository,
+            "default_build": self.default_build,
+            "builds": {
+                b.name: b.serialize() for b in self.builds.values()
+            },
+        }
+
+        return serialized_component
+
+    def _compute_self_hash(self):
+        to_hash = ""
+        for build_name in sorted(self.builds.keys()):
+            to_hash += self.builds[build_name].build_hash
+
+        commit = self.commit()
+        if commit:
+            to_hash += commit
+
+        return hashlib.sha1(to_hash.encode("utf-8")).hexdigest()
```

Comment:
Can we have in a single place in the whole codebase
```python
def hash(to_hash):
    return hashlib.sha1(to_hash.encode("utf-8")).hexdigest()
```
This makes it easier to change things.
I thought I already commented something along these lines.

Reply:
I did not understand that you wanted to separate the hash computation in its own function, I thought you just wanted to remove lazyness. Makes sense, I've made this change.